xdsl/irdl/declarative_assembly_format.py

```diff
@@ -51,7 +50,7 @@ class ParsingState:
     successors: list[Successor | None | Sequence[Successor]]
     attributes: dict[str, Attribute]
     properties: dict[str, Attribute]
-    variables: dict[str, ConstraintVariableType]
+    context: InferenceContext
```

Comment:
I feel `context` is a slightly ambiguous name but I'm getting the feeling that this is part of a bigger change that may make it make more sense

Reply:
Yeah it's InferenceContext in this PR but will be a ConstraintContext in a future one.